### ISSUE TITLE
More than one Java maven project is not displayed/resolved correctly in Java Projects view of multi root workspace in VS Code

### ISSUE THREAD

**Issue by anazar2001** [redhat-developer/vscode-java#2545]
[client.log](https://github.com/redhat-developer/vscode-java/files/9053696/client.log)
[client_log2.log](https://github.com/redhat-developer/vscode-java/files/9053737/client_log2.log)

Hi,

A couple of days ago noticed that when I add new java project It does not get resolved (classpath, JRE, etc) in Java Projects view. Hence Tests not working and in general it is not a java project for VS Code. If I remove the first project from the workspace then the second becomes a valid project.

I have the same laptop at home which is synced via github account for settings, extensions, etc and home laptop works fine but on small projects.

My work laptop periodically takes very long time to open Java projects (I have normally 5-7) in the workspace. When it can't within 2 min I just do cleanup workspace and after reload it can open, though still takes some time. This got worse over time. I already reinstalled vs code and cleaned all metadata. Did not help.

When project is not recognized as a Java project the tree node in java projects corresponding for this project is not expanding.

Could someone take a look and fix this as it is very annoying.

Thank you,
Andrei.

##### Environment
- Operating System: Darwin arm64 21.1.0
- JDK version: Zulu 8.60.0.21-CA-macos-aarch64
- Visual Studio Code version: .68.1 (https://github.com/microsoft/vscode/commit/30d9c6cd9483b2cc586687151bcbcd635f373630, 2022-06-14T12:48:37.209Z)
- Java extension version: 1.8.0

##### Steps To Reproduce
1. Create a new workspace
2. Import/Create new Java maven project
3. Add to workspace a folder with a another Java maven project
4. Check that the second project can be expanded and you can see packages, JDK, Maven dependencies, etc in Java Projects view.
5. This does not happen for any 'second project'. Simple 'hello world' projects work sometime.
6. If you remove the first project then the second gets displayed/handled properly.
[client.log](https://github.com/redhat-developer/vscode-java/files/9053701/client.log)

[Please attach a sample project reproducing the error]
[Please attach logs](https://github.com/redhat-developer/vscode-java/wiki/Troubleshooting#enable-logging)

##### Current Result

Second, third, etc projects in the Multi root java workspace stopped working 2-5 days ago

##### Expected Result

Multiple Java projects should work in Multi root java workspace

**Comment by fbricon** [comment]
Looks like a regression in 1.8.0. Can you reinstall 1.7.0 and check if it still works?

**Comment by anazar2001** [comment]
It is a proprietory project in Mastercard. Can't share it(( Where can I get 1.7.0?

**Comment by fbricon** [comment]
<img width="627" alt="Screenshot 2022-07-06 at 12 24 06" src="https://user-images.githubusercontent.com/148698/177529604-d18db729-1fd2-40e9-8613-baf091f17961.png">
<img width="868" alt="Screenshot 2022-07-06 at 12 24 20" src="https://user-images.githubusercontent.com/148698/177529610-c48f955e-a6fb-4cb1-94cf-e4526fc81242.png">

**Comment by anazar2001** [comment]
Thank 

> <img alt="Screenshot 2022-07-06 at 12 24 06" width="627" src="https://user-images.githubusercontent.com/148698/177529604-d18db729-1fd2-40e9-8613-baf091f17961.png">
> 
> <img alt="Screenshot 2022-07-06 at 12 24 20" width="868" src="https://user-images.githubusercontent.com/148698/177529610-c48f955e-a6fb-4cb1-94cf-e4526fc81242.png">

Thank you but I am using VS Code on my Macbook and interface looks slightly different - can't see where I can choose a different version. Any idea?

**Comment by fbricon** [comment]
I'm also on mac

https://user-images.githubusercontent.com/148698/177535341-e3f66b3d-2c7c-42be-8959-6c63f8508372.mp4

**Comment by anazar2001** [comment]
> I'm also on mac
> 
>  Jul-06-2022.12-54-21.mp4

My VS Code looks different. Please take a look
<img width="1728" alt="Screenshot 2022-07-06 at 12 14 00" src="https://user-images.githubusercontent.com/13858727/177538146-ea599c17-7e37-4d9c-bc76-7600a2489e3d.png">

**Comment by fbricon** [comment]
Ah right, for some reason Mac M1s get a different build. You can still install 1.7.0 manually from https://github.com/redhat-developer/vscode-java/releases/download/v1.7.0/redhat.java@darwin-arm64-1.7.0.vsix

**Comment by anazar2001** [comment]
How can I install it? I downloaded it. Trying to open it. Mac asks for app, I select VS Code but it tries to open it as a text and asks for what editor to use.

**Comment by anazar2001** [comment]
> How can I install it? I downloaded it. Trying to open it. Mac asks for app, I select VS Code but it tries to open it as a text and asks for what editor to use.

How can I install it? I downloaded it. Trying to open it. Mac asks for app, I select VS Code but it tries to open it as a text and asks for what editor to use.

**Comment by fbricon** [comment]
Need to investigate, possible regression with m2e. Having minimal test projects to reproduce would certainly help

**Comment by anazar2001** [comment]
Hi Fred,

Any idea when this issue will be fixed? Can see that 1.9 still has the bug.
So could not update to it. Now I am 2 versions behind(((

Thanks,
Andrei.

**Comment by yorkish** [comment]
I can confirm that the issue is """resolved""" when I backtrack to version 1.7.0 of the extension. 
Running RockyLinux 8.5 + VsCode 1.70.2

**Comment by anazar2001** [comment]
Yes, it works with 1.7. But it needs to be fixed in the latest version too.
When can it be done? It is clearly a regression.

**Comment by snjeza** [comment]
A related issue - https://github.com/redhat-developer/vscode-java/issues/2647

**Comment by anazar2001** [comment]
Any updates on this? Even downgrade of java extension to 1.7.0 stopped working. Cleaned workspace many times

**Comment by yorkish** [comment]
What changed between 1.7.0 and 1.8.0 that caused this regression ?

**Comment by yorkish** [comment]
So nobody is working with 2 or more java projects in the same workspace or isn't using VS Code to do it ?!

**Comment by anazar2001** [comment]
Actually, I think the issue was fixed intentionally or not. A couple of versions ago. I have 10+ projects in my workspace. Loading is slow but once all is loaded it works.

**Comment by yorkish** [comment]
I'll give it another try

**Comment by snjeza** [comment]
> I'll give it another try

@yorkish Could you provide a project example reproducing the error?
